osu.Game/Database/RealmContextFactory.cs

```diff
@@ -374,17 +385,17 @@ public IDisposable BlockAllOperations()
             if (isDisposed)
                 throw new ObjectDisposedException(nameof(RealmContextFactory));
 
-            if (!ThreadSafety.IsUpdateThread)
-                throw new InvalidOperationException(@$"{nameof(BlockAllOperations)} must be called from the update thread.");
-
-            Logger.Log(@"Blocking realm operations.", LoggingTarget.Database);
-
             try
             {
                 contextCreationLock.Wait();
 
                 lock (contextLock)
                 {
+                    if (!ThreadSafety.IsUpdateThread && context != null)
```

Comment:
The `context != null` check is to just make calling the migration from `OsuGameBase.load()` work, correct? Maybe it should be marked with a removal TODO at some point when it's no longer required?

Reply:
I'm not sure there will be a point it's not required. I think we want to be able to freely use realm during the initial load? Not sure.

Reply:
I suppose there shouldn't be much of an issue with that, yeah.